src/NuGet.Core/NuGet.Protocol/Converters/VersionRangeConverter.cs

```diff
@@ -29,7 +29,8 @@ public class VersionRangeConverter : JsonConverter
         /// <returns>A <see cref="VersionRange" /> object.</returns>
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            return reader.TokenType != JsonToken.Null ? VersionRange.Parse(serializer.Deserialize<string>(reader)) : null;
+            string value = serializer.Deserialize<string>(reader);
```

Comment:
Looks good, do you happen to know why `VersionRangeConverter.WriteJson()` gets an `object`, calls `ToString()`, then `VersionRange.Parse()`, and finally `serializer.Serialize(writer, versionRange.ToString())`.  Is that to make sure its a `VersionRange`?  That method seems wild to me.

Reply:
I have no idea. I assumed this was written before I joined the team, but I was in the team at the time. My name is even in the requested reviewers, although I didn't comment on the PR.

Ultimately it doesn't really matter, because there's no scenario where NuGet will serialize the in-memory data model into JSON, so it's effectively dead code. Just a mandatory API from the base class.